src/Spd.Manager.Licence/BizLicAppMananger.cs

```diff
@@ -112,7 +113,60 @@ public async Task<BizLicAppCommandResponse> Handle(BizLicAppReplaceCommand cmd,
 
     public async Task<BizLicAppCommandResponse> Handle(BizLicAppRenewCommand cmd, CancellationToken cancellationToken)
     {
-        throw new NotImplementedException();
+        BizLicAppSubmitRequest request = cmd.LicenceRequest;
+        if (cmd.LicenceRequest.ApplicationTypeCode != ApplicationTypeCode.Renewal)
+            throw new ArgumentException("should be a renewal request");
+
+        // Validation: check if original licence meet renew condition.
+        LicenceListResp originalLicences = await _licenceRepository.QueryAsync(
+            new LicenceQry() { LicenceId = request.OriginalLicenceId },
+            cancellationToken);
+        if (originalLicences == null || !originalLicences.Items.Any())
+            throw new ArgumentException("cannot find the licence that needs to be renewed.");
+        LicenceResp originalLic = originalLicences.Items.First();
+
+        // Check Renew your existing application before it expires, within 90 days of the expiry date.
+        if (DateTime.UtcNow < originalLic.ExpiryDate.AddDays(-Constants.LicenceWith123YearsRenewValidBeforeExpirationInDays).ToDateTime(new TimeOnly(0, 0))
```

Comment:
Have you check the specific requirment for this?

Reply:
I did, in confluence: https://justice.gov.bc.ca/wiki/display/SPDBT/Business+Licence

**Section: Licence Term**